File: 2016/07/puzzle_01.exs
{:ok, contents} = File.read("./input.txt")

abba_pattern = '((.)(?!\\2)(.))\\3\\2'

contents
  |> String.split("\n")
  |> Enum.filter(fn(line) ->
      Regex.match?(~r/#{abba_pattern}/, line) &&
      !Regex.match?(~r/\[[^\[\]]*#{abba_pattern}[^\[\]]*\]/, line)
    end)
  |> Enum.count
  |> IO.puts

File: 2019/08/puzzle_01.rb
pixels = File.read("input.txt").chomp.split("").map(&:to_i)

min_0_layer = pixels.each_slice(25*6).min {|layer_a, layer_b| layer_a.count(0) <=> layer_b.count(0)}

puts min_0_layer.count(1) * min_0_layer.count(2)
